src/open-feature.ts

```diff
@@ -0,0 +1,59 @@
+import { OpenFeatureClient } from './client';
+import { NOOP_PROVIDER } from './no-op-provider';
+import { Client, EvaluationContext, EvaluationLifeCycle, FlagValue, Hook, Provider } from './types';
+
+// use a symbol as a key for the global singleton
+const GLOBAL_OPENFEATURE_API_KEY = Symbol.for('@openfeature/js.api');
+
+type OpenFeatureGlobal = {
+  [GLOBAL_OPENFEATURE_API_KEY]?: OpenFeature;
+};
+const _global = global as OpenFeatureGlobal;
+
+export class OpenFeature implements EvaluationLifeCycle {
+  private _provider: Provider = NOOP_PROVIDER;
```

Comment:
Does the linter enforce an underscore prefix for private properties? If not, it might be nice to add it so names stay consistent.

Reply:
I will look into adding that, I like it too.